src/Model/ModelManager.php

```diff
@@ -336,9 +337,10 @@ public function addIdentifiersToQuery(string $class, BaseProxyQueryInterface $qu
 
             $ands = [];
             foreach ($fieldNames as $posName => $name) {
+                $fieldType = $metadata->getTypeOfField($name);
                 $parameterName = sprintf('field_%s_%s_%d', $prefix, $name, $pos);
                 $ands[] = sprintf('%s.%s = :%s', current($qb->getRootAliases()), $name, $parameterName);
-                $qb->setParameter($parameterName, $ids[$posName]);
+                $qb->setParameter($parameterName, $ids[$posName], $fieldType);
```

Comment:
This will be a bit problematic if the id is for example a DateTime... We need to find a better solution for this. Will also add tests for it.

Reply:
Just tested, since doctrine does not support DateTime as a primary key it is not a problem for that kind of field.